### PR TITLE
Boot the kicked user from the room.

### PR DESCRIPTION
**Pull request by zpChris**
### List of Changes:

- Boot the kicked user from the room, showing them an error message and disconnecting them from the socket.
- Fix issues on the `Socket.ts` file with the `disconnect()` function, and remove unnecessary variables.
- Change frontend `Lobby.tsx` file to use `userId` to compare users rather than `nickname`, in instances where possible (this seemed a more reliable long-term to me, though I'm open to reversing this change).

### Notes:

- The current design of kicking users -- removing them from the room when they are not detected in the user list and disconnecting them from the socket -- means that the kicked user could just rejoin the room if they wanted (as long as the room had not started the game). This is a useful action, but we could also introduce the concept of "banning" users, where there would be a separate "banned users" list; the problem with this is that right now users are only identified by their nickname initially, so in order to appropriately implement this we would have to associate each user with an ID via cookies or similar on their computer.
- Furthermore, the way users are booted if not yet connected to a socket is that if their nickname is not found in the inactive users list when `currentUser` is not yet set, then the user is assumed to have been kicked and is booted. This may require a more rigorous solution in the future, as I could imagine a bug where the user is not booted but the nickname is not transferred properly.
- In the `Socket.ts` file, the code before would reset the `stompClient`, which actually would create an issue with disconnecting from the socket; instead, in the `disconnect()` file we assume that if the socket is connected, then the `stompClient` does not need to be set.
- Additionally, in the `Socket.ts` file, the `connected` variable is not needed as this is emboided in the `stompClient` variable, which has a `connected` field; I thus removed and replaced this in that file.
- I wanted to note that currently in the frontend, the `currentUser` is initially identified and set based on if the `nickname` passed through in `location` matches any of the user's nicknames in the room, then takes the first one and sets that `user` as the `currentUser`. We may eventually want to create the user and pass this through location in the future, so people could potentially join with the same nickname if this is ever wanted in the future.

### Screencast and Images:

[Screencast of Booting a Kicked User (and Edge Cases)](https://www.loom.com/share/0de6e95628b44c67b6213485706f3c59)

**Image of Join Page After Kicked User**
Image of Join Page After Kicked User

**Image of Join Page when User Cannot Be Found in Room's List of Users**
Image of Join Page when User Cannot Be Found in Room's List of Users

**Image of Error on Terminal (Backend) when User is Booted **
_This is an error that **sometimes** pops up in the Terminal when an inactive user is kicked. I'm not sure what is causing this, or under what circumstances this error shows, so I'll leave it for now._
Error on Booted User Terminal

### CODE REVIEW

**Comment by zpChris** [comment]
[Redirect and disconnect user when kicked](https://trello.com/c/RFkGzwro/214-redirect-and-disconnect-user-when-kicked)

File: frontend/src/views/Lobby.tsx [hunk 1]
```diff
@@ -193,12 +253,7 @@ function LobbyPage() {
       getRoom(location.state.roomId)
         .then((res) => {
           setStateFromRoom(res);
-          // Reset the user to hold the ID.
-          res.users.forEach((user: User) => {
-            if (user.nickname === location.state.user.nickname) {
-              setCurrentUser(user);
-            }
-          });
+          updateCurrentUserDetails(res.inactiveUsers);
```
Comment (alankbi) [inline frontend/src/views/Lobby.tsx:256]: nit: could you change this to pass in the list of all users? I'm hoping that doesn't break anything in your PR but I remember that was needed to fix the CMD+SHIFT+T bug

**Review by alankbi — APPROVED** [review]
Approved, just left 1-2 suggestions to take a look at and see if those make sense/are feasible

File: frontend/src/views/Lobby.tsx (hunk 1 above)
Comment (zpChris) [reply frontend/src/views/Lobby.tsx:260]: Yeah no problem -- I'm not certain whether that fixed the error on this page, but it doesn't break my PR so I made the change 👍

File: frontend/src/views/Lobby.tsx [hunk 2]
```diff
@@ -169,9 +215,23 @@ function LobbyPage() {
...
+        room.users.forEach((user) => {
+          if (currentUser.userId === user.userId) {
+            userIncluded = true;
+          }
+        });
+        // If user is no longer present in room, boot the user.
+        if (!userIncluded) {
+          bootKickedUser();
+        }
+      }
```
[the remark this answers is not in the document]
Comment (zpChris) [reply frontend/src/views/Lobby.tsx]: I moved the code from line 223 to line 233 above to `bootKickedUser`, renamed it `conditionallyBootKickedUser` as the `currentUser` must be set and must not be in the room, and updated the relevant dependencies and parameters. Should I be updating this any further @alankbi? I don't see a need to combine this with `updateCurrentUserDetails` as they are performing different functions and the `currentUser` only needs to be set once, whereas the user must be checked whether they were booted on every socket update.

**Review by helenli8181 — APPROVED** [review]
Approved, this looks great! :)

File: frontend/src/views/Lobby.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (zpChris) [reply frontend/src/views/Lobby.tsx]: Confirmed over Messenger with @alankbi that this is OK. 👍

**Pull request merged**